user_mgr.cpp

```diff
@@ -295,9 +295,16 @@ void UserMgr::throwForInvalidGroups(const std::vector<std::string>& groupNames)
     }
 }
 
+/* Notes for restricted priv-operator role:
+ *
+ * The priv-operator role is restricted so you cannot create an operator user
+ * or change an existing user to have the operator role.  However, if there
+ * happens to be a user with the operator role, you are allowed to rename or
+ * delete that user, or change them away from the operator role.
+ */
 void UserMgr::throwForRestrictedPrivilegeRole(const std::string& priv)
```

Comment:
It would also throw here: 
https://github.com/ibm-openbmc/phosphor-user-manager/pull/11/files#diff-1ad65bb42cff8bb43dcb9c709cc9bd656245da4e991b40776456c43f1880743bR435

That is what you want? 
Anytime 
```
void UserMgr::updateGroupsAndPriv(const std::string& userName,
                                  const std::vector<std::string>& groupNames,
                                  const std::string& priv)
```

Reply:
Yes, that is my intention.  I want UserMgr::updateGroupsAndPriv, to throw when:
- throwForRestrictedPrivilegeRole - (new condition) when the NEW role is priv-operator or (existing condition) when the NEW role is priv-oemibmserviceagent
- throwForRestrictedUserPrivilegeRole - (existing condition) when the current user has the priv-oemibmserviceagent role

Reply:
In this way:
- This keeps the restrictions for both using the ServiceAgent role and for changing users who have the service agent role.
- This adds a new restriction for changing to the Operator role, but does not restrict changes to users who have the Operator role.  Note that we should never have role=Operator users in FW1020.00, but we might get some from a customer who moves up to FW1020.next, creates a role=Operator user, then downgrades their BMC to FW1020.00. The ability to manage such use will become important in that case, and that is the code I intended to write.  Specifically, for FW1020.00 to support the following operations for role=Operator users:
    - delete an existing operator user.
    - change an operator user to some other role (Administrator or ReadOnly).